HeterogeneousCore/CUDAUtilities/test/AtomicPairCounter_t.cu

```diff
@@ -0,0 +1,65 @@
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/AtomicPairCounter.h"
+
+#include "HeterogeneousCore/CUDAUtilities/interface/cuda_assert.h"
+
+__global__ void update(AtomicPairCounter *dc, uint32_t *ind, uint32_t *cont, uint32_t n) {
+  auto i = blockIdx.x * blockDim.x + threadIdx.x;
+  if (i >= n)
+    return;
+
+  auto m = i % 11;
+  m = m % 6 + 1;  // max 6, no 0
+  auto c = dc->add(m);
+  assert(c.m < n);
+  ind[c.m] = c.n;
+  for (int j = c.n; j < c.n + m; ++j)
+    cont[j] = i;
+};
+
+__global__ void finalize(AtomicPairCounter const *dc, uint32_t *ind, uint32_t *cont, uint32_t n) {
+  assert(dc->get().m == n);
+  ind[n] = dc->get().n;
+}
+
+__global__ void verify(AtomicPairCounter const *dc, uint32_t const *ind, uint32_t const *cont, uint32_t n) {
+  auto i = blockIdx.x * blockDim.x + threadIdx.x;
+  if (i >= n)
+    return;
+  assert(0 == ind[0]);
+  assert(dc->get().m == n);
+  assert(ind[n] == dc->get().n);
+  auto ib = ind[i];
+  auto ie = ind[i + 1];
+  auto k = cont[ib++];
+  assert(k < n);
+  for (; ib < ie; ++ib)
+    assert(cont[ib] == k);
+}
+
+#include <iostream>
+int main() {
+  AtomicPairCounter *dc_d;
```

Comment:
This would need
```cpp
cms::cudatest::requireDevices();
```
to fix the unit test failure.

Reply:
Thanks, should be fixed now.